onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -0,0 +1,695 @@
+            if (isPowerOfTwo(m_inputDims[m_axis])) {
+                StockhamFFT(inputResource.Get(), outputResource.Get(), commandList.Get());
+            }
+            else {
+                BluesteinZChirp(inputResource.Get(), outputResource.Get(), commandList.Get());
+            }
+            return S_OK;
+        }
+        catch (...)
+        {
+            return E_FAIL;
+        }
+    }
+
+    void StockhamFFT(
+        ID3D12Resource* inputResource,
+        ID3D12Resource* outputResource,
+        ID3D12GraphicsCommandList* commandList)
+    {
+        // Transition resources from common to UAV state
+        D3D12_RESOURCE_BARRIER barriers[2];
+
+        barriers[0] = CD3DX12_RESOURCE_BARRIER::Transition(
+            inputResource,
+            D3D12_RESOURCE_STATE_COMMON,
+            D3D12_RESOURCE_STATE_UNORDERED_ACCESS
+        );
+
+        barriers[1] = CD3DX12_RESOURCE_BARRIER::Transition(
+            outputResource,
+            D3D12_RESOURCE_STATE_COMMON,
+            D3D12_RESOURCE_STATE_UNORDERED_ACCESS
+        );
+
+        commandList->ResourceBarrier(2, barriers);
+
+        // Set the root signature and pipeline state
+        commandList->SetComputeRootSignature(m_rootSignature.Get());
+        commandList->SetPipelineState(m_pipelineState.Get());
+
+        // Each iteration of the below loop represents 1 level in the Stockham DFT
+        // Dispatch in a loop
+        DFTShaderConstants constants = {};
+        constants.DFTIteration = 0;
+        constants.IsInverse = m_isInverse;
+
+        auto resourceLoopList = resourceLoopList_;
+        resourceLoopList[0].Resource = inputResource;
+        resourceLoopList[m_outputIdx].Resource = outputResource;
+
+        for (unsigned index = 0; index < m_numPasses; index++) {
+            auto inIdx = loopRange_.CalculateIndex(index);
+            auto outIdx = loopRange_.CalculateIndex(index + 1);
+
+            auto in = resourceLoopList[inIdx].Resource.Get();
+            std::copy(resourceLoopList[inIdx].Sizes.begin(), resourceLoopList[inIdx].Sizes.end(), constants.InputSizes);
+            std::copy(resourceLoopList[inIdx].Strides.begin(), resourceLoopList[inIdx].Strides.end(), constants.InputStrides);
+
+            auto out = resourceLoopList[outIdx].Resource.Get();
+            std::copy(resourceLoopList[outIdx].Sizes.begin(), resourceLoopList[outIdx].Sizes.end(), constants.OutputSizes);
+            std::copy(resourceLoopList[outIdx].Strides.begin(), resourceLoopList[outIdx].Strides.end(), constants.OutputStrides);
+
+            auto isLastPass = (index == m_numPasses - 1);
+            auto isLastInversePass = isLastPass && m_isInverse;
+            auto dftLength = 1 << m_numPasses;
+            constants.Scale = isLastInversePass ? (1.f / dftLength) : 1.f;
+
+            auto totalElementCount =
+                std::accumulate(constants.OutputSizes,
+                                constants.OutputSizes + _countof(constants.OutputSizes),
```

Comment:
>_countof [](http://example.com/codeflow?start=56&length=8)

`std::size` (avoid MSVC specific intrinsic function, since we want ORT with DML EP to compile with clang too)